source/PlayerInfo.cpp

```diff
@@ -1431,10 +1431,6 @@ void PlayerInfo::Land(UI *ui)
 			++it;
 	}
 
-	// "Unload" all fighters, so they will get recharged, etc.
-	for(const shared_ptr<Ship> &ship : ships)
-		ship->UnloadBays();
```

Comment:
By moving this in the loop below, carriers that can't land on the current planet won't get their fighters recharged even if the carrier landed on another planet. So I would suggest keeping it like it was, unless I'm missing something.